src/fusor/fusion_caller_models.py

```diff
@@ -24,10 +20,16 @@ class Caller(str, Enum):
     GENIE = "GENIE"
 
 
-class JAFFA(BaseModel):
+class FusionCaller(ABC, BaseModel, extra="forbid"):
+    """ABC for fusion callers"""
+
+    type: Caller
+
+
+class JAFFA(FusionCaller):
     """Define parameters for JAFFA model"""
 
-    type: Literal[Caller.JAFFA] = Caller.JAFFA
+    type: Caller = Caller.JAFFA
```

Comment:
So I think we want to restrict this to only be Caller.JAFFA (and in other places), so I think this change can be reverted